clang/test/CheckedCRewriter/basic_checks.c

```diff
@@ -20,9 +21,14 @@ void typd_driver(void) {
   *(b+4) = 0;
 }
 //CHECK: void typd_driver(void) {
-//CHECK-NEXT: wchar_t buf[10];
-//CHECK-NEXT: wchar_t *a = &buf[0];
-//CHECK-NEXT: wchar_t *b = &buf[0];
+//CHECK_NOALL: wchar_t buf[10];
+//CHECK_NOALL: wchar_t *a = &buf[0];
+//CHECK_NOALL: wchar_t *b = &buf[0];
+
+//CHECK_ALL: wchar_t buf _Checked[10];
+//CHECK_ALL: _Ptr<wchar_t> a = &buf[0];
```

Comment:
This is surprising to me: Why would using `-alltypes` figure out that this is a normal `_Ptr` but running without the flag (per the above annotation) not choose `_Ptr`? I would expect `-alltypes` to be adding array annotations where previously they were WILD, but it wouldn't add new `_Ptr` annotations.

Reply:
The way I justified it to myself is that without alltypes, all array pointers are WILD, and so doing &buf[0], the tool propagates that wildness to the pointer, whereas with alltypes, the array is checked, so the index operation of buf[0] is "valid", and therefore `a` must be a valid pointer. I'm guessing this isn't what we want?

Reply:
Hm, that's very interesting. No, I think what you have is right, I didn't think about the case as you've described it. Thanks!